test/back-end.test.js

```diff
@@ -0,0 +1,200 @@
+        expect(response.body.title_url).toEqual('error_not_a_title_in_db');
+        expect(response.body.tomatometer).toBeLessThanOrEqual(0);
+        expect(response.body.avgrating).toBeLessThanOrEqual(0);
+        expect(response.body.numOfReviews).toBeGreaterThanOrEqual(0);
+        expect(response.body.fresh).toBeGreaterThanOrEqual(0);
+        expect(response.body.rotten).toBeGreaterThanOrEqual(0);
+        expect(response.body.randomReview).toBeDefined;
+        done();
+      });
+  });
+
+  test('It should respond to no id with a valid object with invalid data', (done) => {
+    return request(app).get('/cr/tomatometer/')
+    .expect(200)
+    .expect('Content-Type', /json/)
+    .then(response => {
+      expect(response.body.id).toEqual(0);
+      expect(response.body.title_url).toEqual('error_not_a_title_in_db');
+      expect(response.body.tomatometer).toBeLessThanOrEqual(0);
+      expect(response.body.avgrating).toBeLessThanOrEqual(0);
+      expect(response.body.numOfReviews).toBeGreaterThanOrEqual(0);
+      expect(response.body.fresh).toBeGreaterThanOrEqual(0);
+      expect(response.body.rotten).toBeGreaterThanOrEqual(0);
+      expect(response.body.randomReview).toBeDefined;
+      done();
+    });
+});
+
+});
+
+//Testing for MovieInfo Endpoint
+
+describe('Test the MovieInfo endpoint', () => {
+  test('It should access the endpoint and retrieve movieInfo based on an ID', (done)=> {
+    return request(app).get('/m/movieinfo/284054')
+    .then(function (response) {
+      expect(response.body.id).toEqual(284054);
+      expect(response.body.title_url).toEqual('black_panther');
+      expect(response.body.title).toEqual('Black Panther');
+      done();
+    })
+  });
+
+  test('It should access the endpoint and retrieve movieInfo based on a title_url', (done)=> {
+    return request(app).get('/m/movieinfo/stan_and_ollie')
+    .then(function (response) {
+      expect(response.body.id).toEqual(394741);
+      expect(response.body.title_url).toEqual('stan_and_ollie');
+      expect(response.body.title).toEqual('Stan & Ollie');
+      done();
+    })
+  });
+
+  test('It should access the endpoint and retrieve a valid object with invalid data based on an invalid id', (done)=> {
+    return request(app).get('/m/movieinfo/0')
+    .then(function (response) {
+      expect(response.body.id).toEqual(0);
+      expect(response.body.title).toEqual('not a movie in db');
+      expect(response.body.title_url).toEqual('not_a_movie_in_db');
+      done();
+    })
+  });
+
+  test('It should access the endpoint and retrieve a valid object with invalid data based on no id', (done)=> {
+    return request(app).get('/m/movieinfo/')
+    .then(function (response) {
+      expect(response.body.id).toEqual(0);
+      expect(response.body.title).toEqual('not a movie in db');
+      expect(response.body.title_url).toEqual('not_a_movie_in_db');
+      done();
+    })
+  });
+})
+
+
+// Testing for getMovieInfo and Tomatometer Database calls

```

Comment:
Could separate API and database tests into two individual unit tests.

Reply:
to be more clear: one back-end and one as API

Reply:
Still not clear on this you think I should separate the files?  The Database Tests and Endpoint Tests are in seperate describes.